pkg/hive/manager_test.go

```diff
@@ -548,3 +549,52 @@ func TestGetClusterDeployment(t *testing.T) {
 		})
 	}
 }
+
+func TestGetClusterSyncforClusterDeployment(t *testing.T) {
+	fakeNamespace := "aro-00000000-0000-0000-0000-000000000000"
+	doc := &api.OpenShiftClusterDocument{
+		OpenShiftCluster: &api.OpenShiftCluster{
+			Properties: api.OpenShiftClusterProperties{
+				HiveProfile: api.HiveProfile{
+					Namespace: fakeNamespace,
+				},
+			},
+		},
+	}
+
+	cs := &hivev1alpha1.ClusterSync{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ClusterDeploymentName,
+			Namespace: fakeNamespace,
+		},
+	}
+
+	for _, tt := range []struct {
+		name    string
+		wantErr string
+	}{
+		{name: "syncset exists and returned"},
+		{name: "syncset does not exist err returned", wantErr: `clustersyncs.hiveinternal.openshift.io "cluster" not found`},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClientBuilder := fake.NewClientBuilder()
+			if tt.wantErr == "" {
+				fakeClientBuilder = fakeClientBuilder.WithRuntimeObjects(cs)
+			}
+			c := clusterManager{
+				hiveClientset: fakeClientBuilder.Build(),
+				log:           logrus.NewEntry(logrus.StandardLogger()),
+			}
+
+			result, err := c.GetClusterSync(context.Background(), doc)
+			if err != nil && err.Error() != tt.wantErr ||
+				err == nil && tt.wantErr != "" {
+				t.Fatal(err)
+			}
+
+			if result != nil && result.Name != cs.Name && result.Namespace != cs.Namespace {
+				t.Fatal("Unexpected cluster sync returned", result)
+			}
```

Comment:
nit:

we don't have to use Fatal because it doesn't have to exit immediately when it fails.
https://pkg.go.dev/testing#T.Fatal

Reply:
Replaced t.Fatal with t.Error